internal/gitinterface/log_test.go

```diff
@@ -102,4 +99,218 @@ func TestGetCommitsBetweenRange(t *testing.T) {
+		assert.Nil(t, commits)
+	})
+	t.Run("Get commits from non-existent commit", func(t *testing.T) {
+		nonExistentHash := EmptyBlob()
+		commits, err := GetCommitsBetweenRange(repo, nonExistentHash, plumbing.ZeroHash)
+		assert.Nil(t, err)
+		assert.Equal(t, commits, []*object.Commit{})
+	})
+}
+
+func TestGetCommitsBetweenRangeForMergeCommits(t *testing.T) {
+	// Creating a tree with merge commits
+	commitIDs := make([]plumbing.Hash, 0, 6)
+	repo, err := git.Init(memory.NewStorage(), memfs.New())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	refName := plumbing.ReferenceName("refs/heads/main")
+
+	if err := repo.Storer.SetReference(plumbing.NewHashReference(refName, plumbing.ZeroHash)); err != nil {
+		t.Fatal(err)
+	}
+
+	ref, err := repo.Reference(refName, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	emptyBlobHash, err := WriteBlob(repo, []byte{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	treeHashes := createTestTrees(t, repo, emptyBlobHash, 6)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// creating the first commit
+
+	commit := CreateCommitObject(testGitConfig, treeHashes[0], []plumbing.Hash{ref.Hash()}, fmt.Sprintf("Test commit %v", 1), testClock)
+	commitHash, err := WriteCommit(repo, commit)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, commitHash)
+
+	// creating two children from the first commit
+
+	// in the visual, these will be commit 2 and commit 3
+
+	children := createChildrenCommits(t, repo, treeHashes, commit.Hash, 2)
+
+	commitIDs = append(commitIDs, children...)
+
+	// creating a child for commit 2, which in the visual will be commit 4
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[3], []plumbing.Hash{children[0]}, fmt.Sprintf("Test commit %v", 4), testClock)
+
+	commitHash, err = WriteCommit(repo, commit)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, commitHash)
+
+	// creating a merge commit from the two children, which in the visual will be commit 5
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[4], children, fmt.Sprintf("Test commit %v", 5), testClock)
+
+	commitHash, err = WriteCommit(repo, commit)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, commitHash)
+
+	// creating a child for commit 3, which in the visual will be commit 6
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[5], []plumbing.Hash{children[1]}, fmt.Sprintf("Test commit %v", 6), testClock)
+
+	commitHash, err = WriteCommit(repo, commit)
+	if err != nil {
+		t.Fatal(err)
+	}
+	commitIDs = append(commitIDs, commitHash)
+
+	// Git tree with merge commit structure without its commit trees and its values:
+	//
+	//  commit 4       commit 5         commit 6
+	//    │              │  │              │
+	//    └─► commit 2 ◄─┘  └─► commit 3 ◄─┘
+	//            │              │
+	//            └─► commit 1 ◄─┘
+
+	allCommits, err := GetCommitsFromCommitIDs(commitIDs, repo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Run("Test commit 1", func(t *testing.T) {
```

Comment:
```suggestion

	t.Run("Test commit 1", func(t *testing.T) {
```

as a rule of thumb, please insert a newline before every new "block" like a new `t.Run` or whatever.

Reply:
Fixed